pkg/handler/collector/file/file.go

```diff
@@ -0,0 +1,90 @@
+//
+// Copyright 2022 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package file
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/guacsec/guac/pkg/handler/processor"
+)
+
+const CollectorFile = "File"
+
+type file struct {
+	path        string
+	lastChecked time.Time
+}
+
+// RetrieveArtifacts collects the documents from the collector. It emits each collected
+// document through the channel to be collected and processed by the upstream processor.
+// The function should block until all the artifacts are collected and return a nil error
+// or return an error from the collector crashing. This function can keep running and check
+// for new artifacts as they are being uploaded by polling on an interval or run once and
+// grab all the artifacts and end.
+func (f *file) RetrieveArtifacts(ctx context.Context, docChannel chan<- *processor.Document) error {
+	if _, err := os.Stat(f.path); os.IsExist(err) {
+		return fmt.Errorf("path: %s does not exist", f.path)
+	}
+
+	readFunc := func(path string, dirEntry fs.DirEntry, err error) error {
+		// NOTE: Explicitly rethrowing new errors if a particular directory has an error.
+		// If we rethrow the error it kills the whole walk. Still useful to make it explit that we ran into an error.
+		if err != nil {
+			return fmt.Errorf("path: %s is invalid", path)
+		}
+		if dirEntry.IsDir() {
+			return nil
+		}
+		if info, err := dirEntry.Info(); !info.ModTime().After(f.lastChecked) || err != nil {
+			return err
+		}
+
+		blob, err := ioutil.ReadFile(path)
+		if err != nil {
+			return err
+		}
+
+		doc := &processor.Document{
+			Blob: blob,
```

Comment:
need to initialize the DocumentType and FormatType

Reply:
Should GCS collector be doing the same thing: https://github.com/guacsec/guac/blob/main/pkg/handler/collector/gcs/gcs.go#L172-L177

Reply:
ah yes it should @pxp928 , should be set to the unknown enum, since the unknown enum is not the empty string.